src/script/lua_api/l_object.cpp

```diff
@@ -1701,43 +1701,183 @@ int ObjectRef::l_hud_get_hotbar_selected_image(lua_State *L)
+			read_color(L, -1, &skybox_params.bgcolor);
+		lua_pop(L, 1);
 
-	std::vector<std::string> params;
-	if (lua_istable(L, 4)) {
-		lua_pushnil(L);
-		while (lua_next(L, 4) != 0) {
-			// key at index -2 and value at index -1
-			if (lua_isstring(L, -1))
-				params.emplace_back(readParam<std::string>(L, -1));
-			else
-				params.emplace_back("");
-			// removes value, keeps key for next iteration
+		lua_getfield(L, 2, "type");
+		if (!lua_isnil(L, -1))
+			skybox_params.type = luaL_checkstring(L, -1);
+		lua_pop(L, 1);
+
+		lua_getfield(L, 2, "textures");
+		skybox_params.textures.clear();
+		if (!lua_istable(L, -1)) {
+		} else {
+			lua_pushnil(L);
+			while (lua_next(L, -2) != 0) {
+				// Key is at index -2 and value at index -1
+				skybox_params.textures.emplace_back(readParam<std::string>(L, -1));
+				// Removes the value, but keeps the key for iteration
+				lua_pop(L, 1);
+			}
+		}
+		lua_pop(L, 1);
+
+		/*
+		We want to avoid crashes, so we're checking even if we're not using them.
+		However, we want to ensure that the skybox can be set to nil when
+		using "regular" or "plain" skybox modes as textures aren't needed.
+		*/
+
+		if (skybox_params.textures.size() != 6 && skybox_params.textures.size() > 0)
+			throw LuaError("Skybox expects 6 textures!");
+
+		skybox_params.clouds = getboolfield_default(L, 2,
+			"clouds", skybox_params.clouds);
+
+		lua_getfield(L, 2, "sky_color");	
+		if (lua_istable(L, -1)) {
+			lua_getfield(L, -1, "day_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.day_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "day_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.day_horizon);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "dawn_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.dawn_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "dawn_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.dawn_horizon);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "night_sky");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.night_sky);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "night_horizon");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.night_horizon);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "indoors");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.indoors);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "fog_sun_tint");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.sun_tint);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "fog_moon_tint");
+			if (!lua_isnil(L, -1))
+				read_color(L, -1, &skybox_params.moon_tint);
+			lua_pop(L, 1);
+
+			lua_getfield(L, -1, "fog_tint_type");
+			if (!lua_isnil(L, -1))
+				skybox_params.tint_type = luaL_checkstring(L, -1);
+			lua_pop(L, 1);
+
+			// Because we need to leave the "sky_color" table.
 			lua_pop(L, 1);
 		}
-	}
+		getServer(L)->setSky(player, skybox_params);
+		lua_pushboolean(L, true);
```

Comment:
`return 1;` here and dropping the `else` below would make the code and diff easier to read.

Reply:
Are you sure? It'd be out of scope as SkyboxParams isn't used outside of it's functions.

Reply:
Nevermind, I got what you mean, move SkyboxParams outside of the if else statement block.